lib/api.js

```diff
@@ -142,6 +143,8 @@ class Api extends Emittery {
 				runStatus = new RunStatus(selectedFiles.length, null);
 			}
 
+			selectedFiles = scheduler.failingTestsFirst(selectedFiles, this._createCacheDir(), this.options.cacheEnabled);
```

Comment:
We should only create the cache dir when it's needed. Perhaps we need to separate a "get predictable cache dir path" which returns `null` when caching is disabled, and "create the cache dir for this invocation of AVA" which uses the same path, predictable or temporary. What do you think?

Reply:
For the moment ava the local cache inside the folder `node_modules/cache/ava` is only used for this function. When we want to go for cache dir in temporary directory, we create a hash of the project-dir (like sha-256 ...) and create the directory from the hash. This would be predictable.